<launch>

  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav)" />
  <arg name="name" default="" />
  <!-- <arg name="portname" default="/dev/MRS_MODULE1" /> -->
  <arg name="portname" default="/dev/ttyUSB0" />
  <!-- baudrate: 9600 19200 38400 57600 115200 230400 460800 500000 576000 921600-->
  <arg name="baudrate" default="9600" /> 
  <arg name="profiler" default="$(optenv PROFILER false)" />

  <!-- will it run using GNU debugger? -->
  <arg name="DEBUG" default="false" />
  <arg unless="$(arg DEBUG)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg DEBUG)" name="launch_prefix_debug" value="debug_roslaunch"/>

  <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="standalone" default="true" />
  <arg name="manager" default="$(arg UAV_NAME)_bacaprotocol_manager" />
  <arg name="n_threads" default="8" />
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

  <group ns="$(arg UAV_NAME)">

    <!-- launch the nodelet -->
    <node pkg="nodelet" type="nodelet" name="serial_estop" args="$(arg nodelet) estop/Estop $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">

      <param name="uav_name" type="string" value="$(arg UAV_NAME)"/>

      <rosparam file="$(find mrs_serial)/config/estop.yaml" />

      <param name="enable_profiler" type="bool" value="$(arg profiler)" />
      <param name="baudrate" type="int" value="$(arg baudrate)" />
      <param name="portname" value="$(arg portname)"/>
      <param name="use_timeout" value="false"/>

      <!-- Publishers -->
      <remap from="~profiler" to="profiler" />
      <remap from="~baca_protocol_out" to="~received_message" />

        <!-- Subscribers -->
      <remap from="~raw_in" to="~send_raw_message" />
      <remap from="~control_manager_in" to="control_manager/diagnostics" />

        <!-- Services -->
      <remap from="~eland_out" to="control_manager/eland" />
      <remap from="~set_leds_out" to="serial_led/set_led" />
      <remap from="~set_ouster_out" to="serial_led/set_ouster" />

    </node>

  </group>

</launch>
